base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -347,15 +364,62 @@ public String getText(Object element) {
 			DISPLAY.asyncExec(() -> {
 				optionalScriptFilePath.ifPresentOrElse(
 						scriptFilePath -> {
-							openFileDialog(scriptFilePath);
+							openFileDialog("Script Generated", "Script", scriptFilePath);
 						},
 						() -> {
 							MessageDialog.openWarning(DISPLAY.getActiveShell(), "Error", "Failed to generate the script");
 						}
 				);
 			});
 		};
-
+		
+	
+	/**
+	 * Listen for generated scripts and display the correct dialog (error or success and open file).
+	 */
+	private PropertyChangeListener generatedDataExchangeFileListener = evt -> {
+			@SuppressWarnings("unchecked")
+			Optional<String> optionalScriptFilePath = Optional.class.cast(evt.getNewValue());
+			DISPLAY.asyncExec(() -> {
+				optionalScriptFilePath.ifPresentOrElse(
+						scriptFilePath -> {
+							openFileDialog("Data File Generated", "Data file", scriptFilePath);
+						},
+						() -> {
+							MessageDialog.openWarning(DISPLAY.getActiveShell(), "Error", "Failed to generate data file");
+						}
+				);
+			});
+		};
+		
+		/**
+		 * Listen for all the data files that has been read and is ready to be displayed to the user for selection.
+		 */
+		private PropertyChangeListener loadDataFileListener = evt -> {
```

Comment:
Shouldn't need this after it becomes a getter

Reply:
Removed.